ResearchKit/Common/ORKStep.m

```diff
@@ -59,6 +59,12 @@ - (Class)stepViewControllerClass {
     return [[self class] stepViewControllerClass];
 }
 
+- (instancetype)copyWithIdentifier:(NSString *)identifier {
+    ORKStep *step = [self copy];
+    step->_identifier = identifier;
```

Comment:
Can you add `ORKThrowInvalidArgumentExceptionIfNil(identifier);` here to prevent a nil `identifier`?
And do `step->_identifier = [identifier copy];`?